### DISCUSSION TITLE
will not get the connection with putty to work...!?

### DISCUSSION THREAD

**Discussion by Tom-246** [tjko/fanpico#124] [Q&A] [answered]
Hi!

I'm trying to get my fanpico 804D v1.2 up and running but do have some trouble I cannot get fixed. If there is someone to give me a helping hand I would appreciate it very much.

I was able to update the firmware to the latest version (v1.7.1). 

#1 Using bootsel + reset I cannot get the fanpico to be shown as an com-connection with my Win11-system. I have to load the MicroPython UF2 file every time. Is this the way to do it?

#2 Having a com-port i use putty to connect to the fanpico. 
checking the connection with '*IDN?' gives an error:

*IDN?
Traceback (most recent call last):
  File "<stdin>", line 1
SyntaxError: invalid syntax

There is a mistake but I don't know what it is...

Kind regards,

**Comment by tjko** [comment, marked as answer]
> #1 Using bootsel + reset I cannot get the fanpico to be shown as an com-connection with my Win11-system. I have to load the MicroPython UF2 file every time. Is this the way to do it?
>

Pressing ```bootsel``` + ```reset``` (press and hold bootsel, then press and release reset) puts Pico into USB Mass Storage mode for uploading firmware (.UF2 file).  In this mode there is no serial (com) connection, but pico appears as a usb disk drive.

FanPico does not use MicroPython. When unit is in USB Mass Storage mode, you'd want to copy appropriate (for example "fanpico-0804D-pico_w.uf2") to the drive that appears to upgrade FanPico to a new firmware version.

 
> #2 Having a com-port i use putty to connect to the fanpico. checking the connection with '*IDN?' gives an error:
> 
> *IDN? Traceback (most recent call last): File "", line 1 SyntaxError: invalid syntax

It would seem like you have loaded MicroPython into the Pico (and not the fanpico firmware UF2 file).

When FanPico firmware is installed, then unit appears as USB serial device/port when it boots up normally (withouth bootsel button pressed when unit is powered on).   (https://github.com/tjko/fanpico/wiki/FanPico-User-Manual#connecting-to-fanpico-console)

**Comment by Tom-246** [comment]
Thank you for your help.

It's working now.
